api/service/legacysync/syncing.go

```diff
@@ -844,7 +905,7 @@ func (ss *StateSync) generateNewState(bc *core.BlockChain, worker *worker.Worker
 		if block == nil {
 			break
 		}
-		err = ss.UpdateBlockAndStatus(block, bc, worker, false)
+		err = ss.UpdateBlockAndStatus(block, bc, true, true)
```

Comment:
why turn verifyAllSeal to true here? then it will verify signature for all blocks, isn't it slow?

Reply:
Here is only a few block to be updated in this case compared to commonBlocks. These blocks is a part of REGISTER / PUSH flow. The logics are:

1. DNS server got a new block in consensus. Block is then pushed to `VerifiedNewBlock` (consensus/consensus_v2.go:712)
2. DNS server send NEWBLOCK request with the new block (with sig) to registered sync peers. (node/node_syncing.go:383)
3. Client node receive NEWBLOCK request, and add blocks to `peerConfig.newBlocks` (api/service/legacysync/syncing.go:253)
4. Blocks added in last step is pulled from `getMaxConsensusBlockFromParentHash` (api/service/legacysync/syncing.go:904), and being handled here.

So blocks updated here are actually small in number compared to commonBlocks. Not a very critical performance issue I think. On the other hand, this enforce signature check will also flush the block's commit sig to db, which ensures that the last block synced from dns sync is committed with a commit sig for consensus to use.

Reply:
cool